### ISSUE TITLE
how to do mix debug with python and c++

### ISSUE THREAD

**Issue by lordrebel** [WebFreak001/code-debug#407]
could you provide a small demo for debugging python mixed with c++ project?

If submitting a bug please make sure

- [x] If you are using gdb
  - [x] `gdb --version` >= 7.7.1
  - [x] it works on the command line with `gdb`
  - [ ] `cwd` and `target` are properly set
- [ ]  If you are using lldb
  - [ ] `lldb --version` >= 3.7.1
  - [ ] it works on the command line with `lldb-mi`
  - [ ] `cwd` and `target` are properly set

Screenshots are helpful but not required

**Comment by WebFreak001** [comment]
if it works on the CLI, it should work the same in vscode - what issues are you experiencing?

**Comment by lordrebel** [comment]
> if it works on the CLI, it should work the same in vscode - what issues are you experiencing?

hi， I am newer with native-debugger, and python/c++ mix debug,I following the [tutoiral](https://nadiah.org/2020/03/01/example-debug-mixed-python-c-in-visual-studio-code/)(that based on cpptools which I dont want to use) and I found It looks like code-debug cannot recogonize command `"processId": "${command:pickProcess}",` so I stucked here...

**Comment by WebFreak001** [comment]
ah pickProcess is not implemented in code-debug, you'll need to manually put in the PID, see [README](https://github.com/WebFreak001/code-debug#attaching-to-existing-processes)

```json
"request": "attach",
"executable": "./bin/python", <-- taken and translated from your tutorial link
"target": "4285" <-- your PID here
```

**Comment by lordrebel** [comment]
in my mind， the process of python is launched by the configue in `launch.json`:
![image](https://github.com/WebFreak001/code-debug/assets/12432469/6cd4f57f-f08b-469c-b6ca-737958b7ca9f)
So, I think I Cannot get the PID before start debug, do you have good idea to work around this?

**Comment by WebFreak001** [comment]
pickProcess doesn't take the PID before starting, it also only works with running processes, so just insert a long sleep or something in your python script to give you time to pick it.

**Comment by lordrebel** [comment]
but the python debug process is launched by `launch.json` right?
![image](https://github.com/WebFreak001/code-debug/assets/12432469/3a88cff0-1b44-4277-a412-a4279ac736c4)

**Comment by WebFreak001** [comment]
yes, once it's launched you look up the PID of the python process that was just launched

**Comment by lordrebel** [comment]
so, my question is " launch the python debug process through the vscode `launch.json` then record the PID to the `lauch.json` maunally and it can worked?"

**Comment by WebFreak001** [comment]
yes that should work, give it a try

**Comment by lordrebel** [comment]
vscode error is: `Failed ro attach: Argument required(process-id to attach).(from target-attach)`

**Comment by WebFreak001** [comment]
you need to insert the PID in the `target` argument, the extension won't just magically know which python process to debug

**Comment by lordrebel** [comment]
yes, but the python process is launched by the `start debugging` button with the configuration in  `launch.json`

**Comment by lordrebel** [comment]
so I cant know the process id before I click the `start debugging` right?

**Comment by WebFreak001** [comment]
you need to be starting two debugging processes, you need to start the python debugger first, then take the PID of it, then start the native debug debugger

In the tutorial you posted it's the same, just taking the PID of it will present you with a list of processes instead of needing to manually copy it from a command

**Comment by WebFreak001** [comment]
see Step 5 in your tutorial, it's almost exactly the same procedure

**Comment by GitMensch** [comment]
Could you add a pickProcess command, possibly with an optional filter (which in this case would be python), please?

**Comment by WebFreak001** [comment]
good idea, I wanted it for a while myself, although I have mostly been using CodeLLDB recently, which worked better for my environment than GDB and also already has a pickProcess command in it.

Still occasionally using this extension and also supporting it in another extension that generates debug configurations automatically, would enable feature parity with CodeLLDB in that regard.
